src/pages/ChoosePersona/ChoosePersona.tsx

```diff
@@ -0,0 +1,208 @@
+
+const ShareButton = styled(StyledButton)`
+  margin-top: auto;
+`;
+
+const CaruouselItem = styled.div`
+  padding: 0 20px;
+`;
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 70vh;
+`;
+
+const callPersonaMutation = (
+  action: string,
+  personaId: string,
+  selectedPersonaId: string
+): Promise<ExecutionResult<RecommendPersonaResponse>> | null => {
+  const recommendPersona = partial(client.mutate, { mutation: RECOMMEND_PERSONA });
+  switch (action) {
+    case 'saved':
+      return null;
+    case 'recommend':
+      return recommendPersona({
+        variables: {
+          recommendedPersonaUuid: personaId,
+          personaUuid: selectedPersonaId,
+        },
+      });
+    default:
+      return null;
+  }
+};
+
+const callSpotMutation = (
+  action: string,
+  spotId: string,
+  selectedPersonaId: string
+): Promise<ExecutionResult<RecommendSpotResponse | ParticipateResponse | AddManagerResponse>> | null => {
+  const recommendSpot = partial(client.mutate, { mutation: RECOMMEND_SPOT });
+  const participate = partial(client.mutate, { mutation: PARTICIPATE });
+  const addMenager = partial(client.mutate, { mutation: ADD_MENAGER });
+  switch (action) {
+    case 'saved':
+      return null;
+    case 'recommended':
+      return recommendSpot({
+        variables: {
+          recommendedPersonaUuid: spotId,
+          personaUuid: selectedPersonaId,
+        },
+      });
+    case 'participant-joined':
+      return participate({
+        variables: {
+          personaId: selectedPersonaId,
+          spotId,
+        },
+      });
+    case 'manager-joined':
+      return addMenager({
+        variables: {
+          personaId: selectedPersonaId,
+          spotId,
+        },
+      });
+    default:
+      return null;
+  }
+};
+
+export const ChoosePersona: FC = () => {
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const { loading, data } = useQuery<GetPersonaType>(GET_PERSONAS);
+  const [setDefaultPersona] = useMutation<SetDefaultPersonaResponse>(SET_DEFAULT_PERSONA);
+  const { user } = useUserContext();
+  const [defaultPersonaUuid, setDefaultPersonaUuid] = useState(user?.defaultPersona);
+  const { pathname } = useLocation();
+  const carousel = useRef<AntCarousel>(null);
+
+  useEffect(() => {
+    if (data?.userPersonas) {
+      const defaultPersonaIndex = data.userPersonas.findIndex(persona => persona.uuid === defaultPersonaUuid);
+      const goToIndex = defaultPersonaIndex === -1 ? 0 : defaultPersonaIndex;
+      setCurrentSlide(goToIndex);
+      carousel.current?.goTo(goToIndex);
+    }
+  }, [data, defaultPersonaUuid]);
+
+  const handleShare = (): void => {
+    const pathnameArray = getPathnameArray(pathname);
+    if (!data) {
+      return;
+    }
+    if (pathnameArray.includes('persona')) {
+      // second element in action type, fourth is action id
+      callPersonaMutation(pathnameArray[2], pathnameArray[4], data.userPersonas[currentSlide].uuid);
```

Comment:
Maybe it would be good to create a variable from [2] and [4] element to name it instead of commenting?